misc-scripts/xref_mapping/XrefParser/MGIParser.pm

```diff
@@ -23,82 +23,57 @@ use strict;
 use warnings;
 use Carp;
 use DBI;
+use Text::CSV;
 
-use base qw(XrefParser::BaseParser);
+use parent qw(XrefParser::BaseParser);
 
 sub run {
 
   my ($self, $ref_arg) = @_;
   my $source_id    = $ref_arg->{source_id};
   my $species_id   = $ref_arg->{species_id};
   my $files        = $ref_arg->{files};
-  my $verbose      = $ref_arg->{verbose};
-  my $dbi          = $ref_arg->{dbi};
-  $dbi = $self->dbi unless defined $dbi;
+  my $verbose      = $ref_arg->{verbose} || 0;
+  my $dbi          = $ref_arg->{dbi} || $self->dbi;
 
   if((!defined $source_id) or (!defined $species_id) or (!defined $files) ){
     croak "Need to pass source_id, species_id and files as pairs";
   }
-  $verbose |=0;
 
-  my $file = @{$files}[0];
+  my $file = shift @{$files};
 
   my $file_io = $self->get_filehandle($file);
   if ( !defined $file_io ) {
     print STDERR "ERROR: Could not open $file\n";
     return 1;    # 1 is an error
   }
-  my %label;
-  my %version;
-  my %description;
-  my %accession;
-
-  my $sql = 'select source_id from source where name like "MGI" and priority_description like "descriptions"';
-  my $sth = $dbi->prepare($sql);
-  
-  $sth->execute();
-  my ($mgi_source_id);
-  $sth->bind_columns(\$mgi_source_id);
-  my @arr;
-  while($sth->fetch()){
-    push @arr, $mgi_source_id;
-  }
-  $sth->finish;
-  
-  $sql = "select accession, label, version,  description from xref where source_id in (".join(", ",@arr).")";
-
-  $sth = $dbi->prepare($sql);
-  $sth->execute();
-  my ($acc, $lab, $ver, $desc);
-  $sth->bind_columns(\$acc, \$lab, \$ver, \$desc);
-  while (my @row = $sth->fetchrow_array()) {
-    if(defined($desc)){
-      $accession{$lab} = $acc;
-      $label{$acc} = $lab;
-      $version{$acc} = $ver;
-      $description{$acc} = $desc;
-    }
-  }
-  $sth->finish;
 
   #synonyms
-  $sql = "insert ignore into synonym (xref_id, synonym) values (?, ?)";
+  my $sql = "insert ignore into synonym (xref_id, synonym) values (?, ?)";
   my $add_syn_sth = $dbi->prepare($sql);    
```

Comment:
Custom SQL alert! Please replace this and all related bits of code with a call to BaseParser::add_to_syn().